libcrux-sha3/src/simd/portable.rs

```diff
@@ -77,25 +100,41 @@ pub(crate) fn load_last<const RATE: usize, const DELIMITER: u8>(
 }
 
 #[inline(always)]
+#[hax_lib::requires(
+    len <= 200 &&
+    start.to_int() + len.to_int() <= out.len().to_int()
+)]
+#[hax_lib::ensures(|_| future(out).len() == out.len())]
 pub(crate) fn store_block<const RATE: usize>(
     s: &[u64; 25],
     out: &mut [u8],
     start: usize,
     len: usize,
 ) {
     let octets = len / 8;
+
+    #[cfg(hax)]
+    let out_len = out.len(); // ghost variable
+
     for i in 0..octets {
-        out[start + 8 * i..start + 8 * i + 8]
-            .copy_from_slice(&get_ij(s, i / 5, i % 5).to_le_bytes());
+        hax_lib::loop_invariant!(|i: usize| out.len() == out_len);
+
+        let value = get_ij(s, i / 5, i % 5);
+        let bytes = value.to_le_bytes();
```

Comment:
Are these changes needed for the proof? If yes, they'd need some explanation to why. If not, we should keep it as is.
Same for the similar change below.